Source/NSDateInterval.m

```diff
@@ -0,0 +1,214 @@
+
+/* Implementation of class NSDateInterval
+   Copyright (C) 2019 Free Software Foundation, Inc.
+   
+   By: Gregory Casamento <greg.casamento@gmail.com>
+   Date: Wed Oct  9 16:24:13 EDT 2019
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02111 USA.
+*/
+
+#include <Foundation/NSDateInterval.h>
+#include <Foundation/NSDate.h>
+#include <Foundation/NSArray.h>
+
+@implementation NSDateInterval
+
+// Init
+- (instancetype)init
+{
+  self = [super init];
+  if(self != nil)
+    {
+      _startDate = nil;
+      _duration = 0.0;
+    }
+  return self;
+}
+
+- (instancetype)initWithStartDate:(NSDate *)startDate 
+                         duration:(NSTimeInterval)duration
+{
+  self = [super init];
+  if(self != nil)
+    {
+      ASSIGNCOPY(_startDate, startDate);
+      _duration = duration;
+    }
+  return self;
+}
+
+- (instancetype)initWithStartDate:(NSDate *)startDate 
+                          endDate:(NSDate *)endDate
+{
+  self = [super init];
+  if(self != nil)
+    {
+      ASSIGNCOPY(_startDate, startDate);
+      ASSIGNCOPY(_endDate, endDate);
+      _duration = [endDate timeIntervalSinceDate: startDate];
+    }
+  return self;  
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  return nil;
+}
+
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+}
+
+- (id) copyWithZone: (NSZone *)zone
+{
+  return [[[self class] allocWithZone: zone]
+           initWithStartDate: _startDate
+                    duration: _duration];
+}
+
+- (void) dealloc
+{
+  RELEASE(_startDate);
+  [super dealloc];
+}
+
+// Access
+- (NSDate *) startDate
+{
+  return _startDate;
+}
+
+- (void) setStartDate: (NSDate *)startDate
+{
+  ASSIGNCOPY(_startDate, startDate);
+}
+
+- (NSDate *) endDate
+{
+  return [_startDate dateByAddingTimeInterval: _duration];
+}
+
+- (void) setEndDate: (NSDate *)endDate
+{
+  _duration = [endDate timeIntervalSinceDate: _startDate];
+}
+
+- (NSTimeInterval) duration
+{
+  return _duration;
+}
+
+- (void) setDuration: (NSTimeInterval)duration
+{
+  _duration = duration;
+}
+
+// Compare
+- (NSComparisonResult) compare: (NSDateInterval *)dateInterval
+{
+  // NSOrderedAscending
```

Comment:
As I already wrote on the original commit, you should store the result of the comparison of the startDates in a local variable and reuse that. Return that value if it is not NSOrderedSame and otherwise compare the durations.